crates/cheatcodes/src/utils.rs

```diff
@@ -71,36 +73,81 @@ impl Cheatcode for ensNamehashCall {
-        Ok(random_number.abi_encode())
+        random_uint(state, None, None)
     }
 }
 
 impl Cheatcode for randomUint_1Call {
     fn apply(&self, state: &mut Cheatcodes) -> Result {
         let Self { min, max } = *self;
-        ensure!(min <= max, "min must be less than or equal to max");
-        // Generate random between range min..=max
-        let exclusive_modulo = max - min;
-        let rng = state.rng();
-        let mut random_number = rng.gen::<U256>();
-        if exclusive_modulo != U256::MAX {
-            let inclusive_modulo = exclusive_modulo + U256::from(1);
-            random_number %= inclusive_modulo;
-        }
-        random_number += min;
-        Ok(random_number.abi_encode())
+        random_uint(state, None, Some((min, max)))
+    }
+}
+
+impl Cheatcode for randomUint_2Call {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { bits } = *self;
+        random_uint(state, Some(bits), None)
     }
 }
 
 impl Cheatcode for randomAddressCall {
     fn apply(&self, state: &mut Cheatcodes) -> Result {
-        let Self {} = self;
-        let rng = state.rng();
-        let addr = Address::random_with(rng);
-        Ok(addr.abi_encode())
+        Ok(DynSolValue::type_strategy(&DynSolType::Address)
+            .new_tree(state.test_runner())
+            .unwrap()
+            .current()
+            .abi_encode())
+    }
+}
+
+impl Cheatcode for randomInt_0Call {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        random_int(state, None)
+    }
+}
+
+impl Cheatcode for randomInt_1Call {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { bits } = *self;
+        random_int(state, Some(bits))
+    }
+}
+
+impl Cheatcode for randomBoolCall {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        Ok(DynSolValue::type_strategy(&DynSolType::Bool)
+            .new_tree(state.test_runner())
+            .unwrap()
+            .current()
+            .abi_encode())
+    }
+}
+
+impl Cheatcode for randomBytesCall {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { len } = *self;
+        ensure!(
+            len <= U256::from(usize::MAX),
+            format!("bytes length cannot exceed {}", usize::MAX)
+        );
+        let mut val = DynSolValue::type_strategy(&DynSolType::Bytes)
+            .new_tree(state.test_runner())
+            .unwrap()
+            .current()
+            .as_bytes()
+            .unwrap_or_default()
+            .to_vec();
+        let required_len = len.to::<usize>();
+        let cur_len = val.len();
+        if cur_len > required_len {
+            // Slice to required length if random bytes length is lower than required length.
+            Ok(val[..required_len].abi_encode())
+        } else {
+            // Fill with zeroes if random bytes length is lower than required length.
+            val.extend(vec![0; required_len - cur_len]);
+            Ok(val.abi_encode())
+        }
```

Comment:
all of this could just be `let mut bytes = vec![0u8; len]; state.rng().fill_bytes(&mut bytes)`

Reply:
changed in https://github.com/foundry-rs/foundry/pull/8882/commits/0a56dfcf2110f55c08bf5fdc86330970b57c07dc